master/internal/command/command_job_service.go

```diff
@@ -47,6 +49,23 @@ func (c *Command) SetJobPriority(priority int) error {
 	if priority < 1 || priority > 99 {
 		return fmt.Errorf("priority must be between 1 and 99")
 	}
+
+	// Returns an error if RM does not implement priority.
+	if smallerHigher, err := c.rm.SmallerValueIsHigherPriority(); err == nil {
+		ok, err := configpolicy.PriorityAllowed(
+			int(c.GenericCommandSpec.Metadata.WorkspaceID),
+			model.NTSCType,
+			priority,
+			smallerHigher,
+		)
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return fmt.Errorf("priority exceeds task config policy's priority_limit")
+		}
+	}
```

Comment:
I think we need to error check here:
`++ else { return fmt.Errorf(...)}`.
AFAICT, the only place where this would error is on Slurm clusters if setting job priority is not supported. If `SetJobPriortiy` does get called with a dispatcher RM, then we should  return that error message here anyway!

Reply:
I'm going to pair with Shreya to test this out, but I'm fairly positive that dispatcher RM does not manage priority. In other words, clusters using dispatcher RM should never execute this code in the first place. 

I think it's ok to attempt to set the task priority even if the RM is not aware of whether priority is `lower > higher` or `higher > lower`. One is a task thing and one is an RM thing.

Reply:
Looking at an actual slurm cluster, there's no priority attached to jobs. Given that, I would prefer to leave the function as-is to avoid unintended consequences - but can also update it if you feel strongly.

Reply:
I'm good with this! Given that `smallerIsHigher` is a boolean, it's not like there's a potential for nil that causes master to panic.
I would argue though that if we don't check for whether or not the error is nil in the code, then the function can probably do without returning the error (in essence, we could just implement dispatcher RM to return `false`).